scripts/test/Muon/elemental_analysis/load_model_test.py

```diff
@@ -0,0 +1,125 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import print_function
+
+import unittest
+
+from mantid.py3compat import mock
+
+from Muon.GUI.ElementalAnalysis.LoadWidget import load_model
+
+
+class LoadModel(unittest.TestCase):
+    def setUp(self):
+        self.model = load_model.LoadModel()
+
+    def test_that_execute_calls_load_run_if_run_not_in_loaded_runs(self):
+        self.model.load_run = mock.Mock()
+        self.model.last_loaded_runs = mock.Mock()
+        self.model.run = 1234
+        self.model.loaded_runs = {1: mock.Mock(), 2: mock.Mock()}
+        self.model.execute()
+
+        self.assertEqual(self.model.load_run.call_count, 1)
+        self.assertEqual(self.model.last_loaded_runs.append.call_count, 0)
+
+    def test_that_execute_adds_run_to_last_loaded_run_if_run_in_loaded_runs(self):
+        self.model.load_run = mock.Mock()
+        self.model.last_loaded_runs = mock.Mock()
+        self.model.run = 1234
+        self.model.loaded_runs = {1234: mock.Mock(), 5678: mock.Mock()}
+        self.model.execute()
+
+        self.assertEqual(self.model.load_run.call_count, 0)
+        self.assertEqual(self.model.last_loaded_runs.append.call_count, 1)
+        self.model.last_loaded_runs.append.assert_called_with(1234)
+
+
+class CoLoadModel(unittest.TestCase):
+    def setUp(self):
+        self.model = load_model.CoLoadModel()
+
+    def test_wipe_co_runs_clears_state(self):
+        self.model.workspace = mock.Mock()
+        self.model.co_runs = [1, 2, 3]
+        self.model.wipe_co_runs()
+
+        self.assertEqual(self.model.workspace, None)
+        self.assertEqual(self.model.co_runs, [])
+
+    def test_that_execute_loads_run_if_not_already_loaded(self):
+        self.model.run = 1234
+        self.model.loaded_runs = {1: mock.Mock(), 2: mock.Mock()}
+        self.model.load_run = mock.Mock(return_value=None)
+
+        # Load failed
+        self.model.execute()
+        self.assertEqual(self.model.load_run.call_count, 1)
+        self.assertTrue(1234 not in self.model.loaded_runs.keys())
+
+        # Load succeeded
+        mock_ws = mock.Mock()
+        self.model.load_run.return_value = mock_ws
+        self.model.execute()
+        self.assertEqual(self.model.load_run.call_count, 2)
+        self.assertTrue(1234 in self.model.loaded_runs.keys())
+        self.assertEqual(self.model.loaded_runs[1234], mock_ws)
+
+    def test_that_run_is_appended_to_last_loaded_run_if_already_loaded(self):
+        self.model.run = 1234
+        self.model.loaded_runs = {1234: mock.Mock(), 5678: mock.Mock()}
+
+        self.assertTrue(1234 not in self.model.last_loaded_runs)
+        self.model.execute()
+        self.assertTrue(1234 in self.model.last_loaded_runs)
```

Comment:
```suggestion
        self.assertTrue(1234 in self.model.last_loaded_runs)
        self.assertTrue(5678 in self.model.last_loaded_runs)
```

Reply:
`5678` will not be added to `self.model.last_loaded_runs`, this is only done to the value in `self.model.run`, in this case `1234`.
https://github.com/mantidproject/mantid/blob/e383f53ca60873572815ef90ea75d596fc71b153/scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_model.py#L35